rfc/entity-caching-v1.md

```diff
@@ -0,0 +1,172 @@
+# RFC Entity Caching
+
+This document describes the design of the entity caching system for Cosmo Router.
+The goal is to provide a high-performance, scalable, and reliable caching system for entities.
+The Router connects to a Cache Service like Redis to store and retrieve entities instead of fetching them from Subgraphs.
+
+This document describes how caching can be implemented by the user in terms of TTL, cache eviction, and cache invalidation.
```

Comment:
I don't think this is called out anywhere in the doc but I'm assuming that the Router will fetch the entire entity from the Subgraph even if the Request is only asking for a few fields ?